Author: Huswaton Hasanah

# Import Libraries

In [3]:
from google.colab import drive
drive.mount('/content/drive')

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Load Dataset

In [4]:
df = pd.read_csv('/content/drive/MyDrive/Rakamin/loan_customer.csv')

# Data Cleansing

## Handling missing values

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 621 entries, 0 to 620
Data columns (total 17 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   loan_id             621 non-null    object 
 1   birth_date          572 non-null    object 
 2   phone_number        569 non-null    float64
 3   gender              608 non-null    object 
 4   married             618 non-null    object 
 5   dependents          605 non-null    object 
 6   education           620 non-null    object 
 7   self_employed       588 non-null    object 
 8   applicant_income    612 non-null    float64
 9   coapplicant_income  621 non-null    float64
 10  loan_amount         621 non-null    int64  
 11  loan_term_month     621 non-null    int64  
 12  loan_term_year      621 non-null    int64  
 13  credit_history      570 non-null    float64
 14  has_credit_card     88 non-null     object 
 15  property_type       620 non-null    object 
 16  loan_sta

Analisis

1. Data terdiri dari 621 baris
2. Tampak beberapa kolom masih memiliki null/missing values (Non-Null Count < jumlah baris)
3. Sepertinya tidak ada issue yang mencolok pada tipe data untuk setiap kolom (sudah sesuai)

## Handling duplicated data

In [6]:
df.isna().sum()

loan_id                 0
birth_date             49
phone_number           52
gender                 13
married                 3
dependents             16
education               1
self_employed          33
applicant_income        9
coapplicant_income      0
loan_amount             0
loan_term_month         0
loan_term_year          0
credit_history         51
has_credit_card       533
property_type           1
loan_status             0
dtype: int64

## Handling Outlier

In [ ]:
'''
print(f'Jumlah baris sebelum memfilter outlier: {len(df)}')

for col in ['loan_id','self_employed', 'coapplicant_income', 'has_credit_card']:
  Q1 = df[col].quantile(0.25)
  Q3 = df[col].quantile(0.75)
  IQR = Q3-Q1
  low_limit = Q1 - 1.5*IQR
  high_limit = Q3 + 1.5*IQR
  df = df[(df[col] >= low_limit) & (df[col] <= high_limit)]

print(f'Jumlah baris setelah memfilter outlier: {len(df)}')
'''

"\nprint(f'Jumlah baris sebelum memfilter outlier: {len(df)}')\n\nfor col in ['loan_id','self_employed', 'coapplicant_income', 'has_credit_card']:\n  Q1 = df[col].quantile(0.25)\n  Q3 = df[col].quantile(0.75)\n  IQR = Q3-Q1\n  low_limit = Q1 - 1.5*IQR\n  high_limit = Q3 + 1.5*IQR\n  df = df[(df[col] >= low_limit) & (df[col] <= high_limit)]\n\nprint(f'Jumlah baris setelah memfilter outlier: {len(df)}')\n"

## Feature Encoding

In [7]:
cats_updated = ['loan_id', 'birth_date', 'phone_number', 'gender','married','dependents','education']

for col in cats_updated:
  print(f'value counts of column {col}')
  print(df[col].value_counts())
  print('---'*10, '\n')

value counts of column loan_id
LP002277    2
LP001990    2
LP002160    2
LP001904    2
LP001310    2
           ..
LP001405    1
LP002941    1
LP002500    1
LP001333    1
LP001013    1
Name: loan_id, Length: 616, dtype: int64
------------------------------ 

value counts of column birth_date
1979-03-23    2
1982-09-18    2
1987-08-07    2
1985-01-19    2
1985-06-09    2
             ..
1988-05-29    1
1992-04-26    1
1985-02-14    1
1978-04-20    1
1980-07-13    1
Name: birth_date, Length: 541, dtype: int64
------------------------------ 

value counts of column phone_number
6.281384e+11    2
6.281153e+11    2
6.281152e+10    2
6.281310e+11    2
6.285279e+11    2
               ..
6.281360e+11    1
6.285603e+10    1
6.281342e+11    1
6.287765e+11    1
6.281391e+11    1
Name: phone_number, Length: 564, dtype: int64
------------------------------ 

value counts of column gender
Male      495
Female    113
Name: gender, dtype: int64
------------------------------ 

value counts of column 

# Business Insight

## Analisis minat pelanggan berdasarkan pada tipe properti

In [8]:
cats_updated = ['property_type']

for col in cats_updated:
  print(f'value counts of column {col}')
  print(df[col].value_counts())
  print('---'*10, '\n')

value counts of column property_type
apartment    235
house        206
studio       179
Name: property_type, dtype: int64
------------------------------ 



berdasarkan data diatas minat pelanggan tertinggi pada properti apertemen

## Analisis pengaruh status perkawinan terhadap jangka waktu meminjam 

In [9]:
cats_updated = ['married','loan_amount','loan_term_month','loan_term_year']

for col in cats_updated:
  print(f'value counts of column {col}')
  print(df[col].value_counts())
  print('---'*10, '\n')

value counts of column married
Yes    403
No     215
Name: married, dtype: int64
------------------------------ 

value counts of column loan_amount
110000    25
120000    21
100000    21
187000    12
160000    12
          ..
58000      1
149000     1
101000     1
59000      1
145000     1
Name: loan_amount, Length: 203, dtype: int64
------------------------------ 

value counts of column loan_term_month
360    533
180     44
480     15
300     13
240      4
84       4
120      3
60       2
36       2
12       1
Name: loan_term_month, dtype: int64
------------------------------ 

value counts of column loan_term_year
30    533
15     44
40     15
25     13
20      4
7       4
10      3
5       2
3       2
1       1
Name: loan_term_year, dtype: int64
------------------------------ 



berdasarkan status perkawinan, pelanggan atau nasabah yang sudah berstatus menikah lebih sering meminjam atau datanya sering muncul selama setahun ini

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 621 entries, 0 to 620
Data columns (total 17 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   loan_id             621 non-null    object 
 1   birth_date          572 non-null    object 
 2   phone_number        569 non-null    float64
 3   gender              608 non-null    object 
 4   married             618 non-null    object 
 5   dependents          605 non-null    object 
 6   education           620 non-null    object 
 7   self_employed       588 non-null    object 
 8   applicant_income    612 non-null    float64
 9   coapplicant_income  621 non-null    float64
 10  loan_amount         621 non-null    int64  
 11  loan_term_month     621 non-null    int64  
 12  loan_term_year      621 non-null    int64  
 13  credit_history      570 non-null    float64
 14  has_credit_card     88 non-null     object 
 15  property_type       620 non-null    object 
 16  loan_sta